core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -354,21 +354,26 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
 
         Set<Link> allLinks = new HashSet<>();
         Set<String> allLinkedContainerNetworks = new HashSet<>();
+
+        List<com.github.dockerjava.api.model.Container> containers = dockerClient.listContainersCmd().exec();
+
+        List<InspectContainerResponse> inspectContainerResponses = containers.stream()
+                .map(container -> dockerClient.inspectContainerCmd(container.getId()).exec())
```

Comment:
Finding the Container from the list containers command response based on container ID is a great idea. The only problem I see is that LinkableContainer doesn't have getContainerId(). I tried adding it, but failed due to FutureContainer. This interface intends to say: I want to link a container by the name of getContainerName(). When you create a FutureContainer in AmbassadorContainer you don't have the ID yet, so it won't work.
Another way to solve this is to define FutureContainer as either name or ID. One way of accomplishing it is having both the new getContainerId() and the existing getContainerName() to return Optional<String>. Then in the method we're discussing now, we can first find the Container from the List<Container> by getContainerId(), otherwise resort to the expensive running inspect per container and then comparing by name using getContainerName().

Another simpler way around it is using parallelStreams.
In my Macbook Pro:
```
MacBook-Pro-3$ time docker inspect priceless_lumiere
...
real	0m0.089s
user	0m0.010s
sys	0m0.013s
```

I ran the same command on a Jenkins Slave in Amazon with instance type of m4.4xlarge, and the result was:
```
real    0m0.525s
user    0m0.041s
sys    0m0.008s
```

Ok, for a CI server, running a 1000 docker containers, it will take 1000 x 0.008s = 8s. This seems quite a lot, but on the other hand,  I don't think this instance type can handle so many docker containers. 
We can shave it off by a factor of number of CPUs by using ".parallelStream()" thus doing this in parallel.  In this case we can achieve 8s / 16 ~ 0.5s. 

So we have two ways of going about this. What do you think?